doc/upgrade-to-2.x.md

```diff
@@ -0,0 +1,475 @@
+
+- removed `BasicTracerProvider#addSpanProcessor(...)` -> use constructor options to the TracerProvider class
+- made `BasicTracerProvider#getActiveSpanProcessor()` private
+- made `BasicTracerProvider#resource` private
+- `BasicTracerProvider` and `NodeTracerProvider` will no longer use the `OTEL_TRACES_EXPORTER` envvar to create exporters -> This functionality already resides in `NodeSDK` (from `@opentelemetry/sdk-node`).
+- `BasicTracerProvider` and `NodeTracerProvider` will no longer use the `OTEL_PROPAGATORS` envvar to create propagators -> This functionality already resides in `NodeSDK` (from `@opentelemetry/sdk-node`).
+- The following internal properties were removed from `BasicTracerProvider`: `_registeredExporters`, `_getSpanExporter`, `_buildExporterFromEnv`
+- The following exports were dropped from `@opentelemetry/sdk-trace-*`:
+  - `EXPORTER_FACTORY` is not used anymore and has been removed
+  - `PROPAGATOR_FACTORY` is not used anymore and has been removed
+  - `ForceFlushState` was intended for internal use and has been removed
+  - the `Tracer` class was unintentionally exported and has been removed
+    - to obtain a `Tracer`, please use `BasicTracerProvider#getTracer()`, `NodeTracerProvider#getTracer()` or `WebTracerProvider#getTracer()`
+    - to reference the `Tracer` type, please use the `Tracer` type from `@opentelemetry/api`
+- removed `BasicTracerProvider#register()` -> use `NodeTracerProvider#register()` or `WebTracerProvider#register()`, or call `trace.setGlobalTracerProvider()` et al manually (see [#5503](https://github.com/open-telemetry/opentelemetry-js/pull/5503))
+
+<br/>
+
+The export of the `Span` class has been removed. It was not intended to be used directly. One should always use methods on a `Tracer` instance (e.g. `startSpan()`) for creating spans.
+
+- `new Span(...)` -> use `tracer.startSpan(...)`
+
+<br/>
+
+The `Span` `parentSpanId` field was replaced by `parentSpanContext`, to adhere to the OTel spec. [#5450](https://github.com/open-telemetry/opentelemetry-js/pull/5450)
+
+- `span.parentSpanId` -> `span.parentSpanContext?.spanId`
+
+<br/>
+
+As mentioned above in the "core" section, `InstrumentationLibrary` has been changed to `InstrumentationScope`:
+
+- `ReadableSpan.instrumentationLibrary` -> `ReadableSpan.instrumentationScope`
+
+<br/>
+
+When bogus data is set on `OTEL_TRACES_SAMPLER`, the SDK now uses `ParentBasedAlwaysOnSampler` rather than `AlwaysOnSampler`, per [spec](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration).
```

Comment:
```suggestion
When invalid data is set on `OTEL_TRACES_SAMPLER`, the SDK now uses `ParentBasedAlwaysOnSampler` rather than `AlwaysOnSampler`, per [spec](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration).
```